superset/assets/src/components/TableSelector.jsx

```diff
@@ -170,13 +170,8 @@ export default class TableSelector extends React.PureComponent {
       this.setState({ tableName: '' });
       return;
     }
-    const namePieces = tableOpt.value.split('.');
-    let tableName = namePieces[0];
-    let schemaName = this.props.schema;
-    if (namePieces.length > 1) {
-      schemaName = namePieces[0];
-      tableName = namePieces[1];
-    }
+    const tableName = tableOpt.value;
+    const schemaName = this.props.schema;
```

Comment:
I think if/when using `TABLE_NAMES_CACHE_CONFIG`, the logic that got removed will be needed.

Reply:
I tested this more thoroughly; it seems table name caching isn't affected, however BigQuery table preview did break due to there being a query of the form ``SELECT ... FROM `schema`.`schema.table` ``. OTOH e.g. the Postgres inspector doesn't append schema to table name. This got me thinking; instead of handling removal of schema from table name in the frontend, shouldn't this be done in the backend? I added an override to the BigQuery spec based on `BaseEngineSpec.get_table_names` method  to remove the schema name after inspection, and it worked perfectly (building on the PR branch):

```python
    @classmethod
    def get_table_names(cls, inspector, schema):
        tables = inspector.get_table_names(schema)
        return sorted([table.split('.')[1] for table in tables])
``` 
This has the added benefit of removing the unnecessary schema name from the table dropdown, and would make it possible to customize this behavior per engine.

Reply:
@villebro Thanks for your help.  I would agree that the table handling/splitting should happen on the backend, so that way custom parsing (such as what I had to do for Drill) could all happen on in the backend.
@mistercrunch, does this complete the PR?